src/controllers/imap.js

```diff
@@ -0,0 +1,70 @@
+import express from 'express';
+import BaseAPIController from './BaseAPIController';
+import ImapProvider from '../providers/ImapProvider';
+import db from '../db';
+
+export class ImapController extends BaseAPIController {
+
+    /* Controller for Save Imap Data  */
+    save = (req, res, next) => {
+        ImapProvider.save(this._db.Imap, req.body, req.imapAccess)
+            .then((data) => {
+                this._db.Imap.save(data)
+                    .then(res.json.bind(res))
+                    .catch(this.handleErrorResponse.bind(null, res));
+            })
+            .catch(this.handleErrorResponse.bind(null, res))
+    }
+
+    /*Get Imapp data using id*/
+    getByID = (req, res, next, id) => {
+        this._db.Imap.findById(id)
+            .then((data) => {
+                if (data) {
+                    req.imapData = data
+                    next()
+                } else {
+                    throw new Error("Invalid Id")
+                }
+            })
+            .catch(this.handleErrorResponse.bind(null, res));
+
+    }
+
+    /*Imap data Update*/
+    update = (req, res, next) => {
+        ImapProvider.save(this._db.Imap, req.body, req.imapAccess)
+            .then((data) => {
+                this._db.Imap.update(data, { where: { id: req.params.id } })
+                    .then(res.json.bind(res))
+                    .catch(this.handleErrorResponse.bind(null, res));
```

Comment:
there is no need to write .catch two times. 

see the previous project zip file. how its done there